qt-models/divelistmodel.h

```diff
@@ -12,11 +12,21 @@ class DiveListSortModel : public QSortFilterProxyModel
 	Q_OBJECT
 public:
 	DiveListSortModel(QObject *parent = 0);
+	void setSourceModel(QAbstractItemModel *sourceModel);
 	Q_INVOKABLE void addAllDives();
 	Q_INVOKABLE void clear();
 public slots:
 	int getDiveId(int idx);
 	int getIdxForId(int id);
+	void setFilter(QString f);
+	void resetFilter();
+	int shown();
+protected:
+	bool filterAcceptsRow(int source_row, const QModelIndex &source_parent) const;
+private:
+	std::vector<bool> filteredRows;
```

Comment:
Not important, but note that `std::vector<bool>` has a design mistake which turns it into a bitfield (!). Thus, it tends to be much less efficient than one might expect.

Reply:
Oh, so I should make this a std::vector<char> or something like that?

Reply:
The reason I'm asking, in case that's not obvious... the main reason to "roll my own" implementation of the filter was performance, so I really wouldn't want to to then use a poor implementation of a vector and lose some of the advantage I'm trying to get...

Reply:
> Oh, so I should make this a std::vector or something like that?

Out of principle, probably yes. Realistically, the microsecond more-or-less will not be perceivable.